src/data-structures/Bytes.huff

```diff
@@ -0,0 +1,114 @@
+#define macro CONCAT_MEMORY(mem_ptr1, mem_ptr2) = takes(2) returns(1) {
```

Comment:
Note that the `<macro_name>(<args>)` arguments are compile-time immediates. This means `mem_ptr1` and `mem_ptr2` _must_ be known at compile time. This may not be known at compile time when dealing with dynamic byte arrays from external sources.

The `takes(n)` indicates how many values (minimum) are expected to be on the stack at the start of the macro's execution. In the current implementation, there are no values on the stack at the macro's start. The `(mem_ptr1, mem_ptr2)` simply declares immediate values.

Consider making the memory pointers dynamic (on the stack) to be interpreted at runtime and change

`#define macro CONCAT_MEMORY(mem_ptr1, mem_ptr2) = takes(2) returns(1)`
to
`#define macro CONCAT_MEMORY() = takes (2) returns (1)`

Reply:
Make sense! Refactored it